node-core/node/node.go

```diff
@@ -73,7 +73,17 @@ func (n *node) Start(
 	}
 
 	// Wait for those aforementioned exit signals.
-	return g.Wait()
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	// Stopp each service allowing them the exit gracefully.
+	if err = n.registry.StopAll(); err != nil {
+		return err
+	}
```

Comment:
should we defer here? So to properly stop all in any case?

Reply:
On partial startup of services I now call StopAll so make sure everything was cleaned up properly.